spec/workers/web/push_notification_worker_spec.rb

```diff
@@ -23,26 +23,39 @@
 
   describe 'perform' do
     before do
-      allow(subscription).to receive_messages(contact_email: contact_email, vapid_key: vapid_key)
-      allow(Web::PushSubscription).to receive(:find).with(subscription.id).and_return(subscription)
+      Rails.configuration.x.vapid_private_key = vapid_private_key
+      Rails.configuration.x.vapid_public_key = vapid_public_key
+      Setting.site_contact_email = contact_email
```

Comment:
I am not sure if this is needed here or if we already have something in place to make this unnecessary, but since I have been bitten by this in the past I just wanted to mention it:

When I change global configuration in a `before` block I always add an `after` block that either rolls back the change or at least `nil`s the values. Otherwise it might introduce unwanted side-effects and/or make tests order dependent.

Reply:
Totally agreed on general pattern to restore any globals to original value in hooks.

In this case, its a db backed value and gets wrapped in the example transaction and rolled back. See https://github.com/mastodon/mastodon/pull/28425

Reply:
OK, that applies to `Setting.site_contact_email`, but does it also apply to the `Rails.configuration.x` values?

Reply:
Oh, ha! -- yes, good point. Added a hook for those.

Reply:
Brilliant, thanks!